core/lib/spree/core/controller_helpers/locale.rb

```diff
@@ -28,6 +28,10 @@ def set_fallback_locale
           Spree::Locales::SetFallbackLocaleForStore.new.call(store: current_store)
         end
 
+        def default_locale
```

Comment:
@damianlegawiec Shouldn't we assign the `default_locale` to  `I18n.default_locale` here: https://github.com/spree/spree/blob/main/core/lib/spree/core/controller_helpers/locale.rb#L21?

If a store has the `default_locale` set to `pl` then the `current_locale` will also be `pl` (unless picked otherwise). So we're still not using any translations and the `I18n.default_locale` should also be `pl`.